doc/api/process.md

```diff
@@ -1881,6 +1881,229 @@ a code.
+but also let the object be garbage collected if it is no longer being used.
+
+Eg: you can register an object that contains a buffer, you want to make sure that buffer is released
+when the process exit, but if the object is garbage collected before the process exit, we no longer
+need to release the buffer, so in this case we just remove the callback from the finalization registry.
+
+Caveats: This function relies on the `FinalizationRegistry` API, which the
+authors of this API said to "avoid where possible" because the nature of
+how the garbage collector works differently between engines, there's no
+specific behavior guarantee by the specification. In their words:
+"Developers shouldn't rely on cleanup callbacks for essential program
+logic. Cleanup callbacks may be useful for reducing memory usage across
+the course of a program, but are unlikely to be useful otherwise". You
+can read more about the caveats at [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/FinalizationRegistry#avoid_where_possible)
+or in the [proposal WeakRefs](https://github.com/tc39/proposal-weakrefs?tab=readme-ov-file#a-note-of-caution).
+
+```cjs
+const { finalization, stdout } = require('node:process');
+
+// Please make sure that the function passed to finalization.register()
+// does not create a closure around unnecessary objects.
+function onFinalize(obj, event) {
+  // You can do whatever you want with the object
+  obj.dispose();
+}
+
+function setup() {
+  // This object can be safely garbage collected,
+  // and the resulting shutdown function will not be called.
+  // There are no leaks.
+  const myDisposableObject = {
+    dispose() {
+      // Free your resources synchronously
+      stdout.write('disposed.\n');
+    },
+  };
+
+  finalization.register(myDisposableObject, onFinalize);
+}
+
+setup();
+```
+
+```mjs
+import { finalization, stdout } from 'node:process';
+
+// Please make sure that the function passed to finalization.register()
+// does not create a closure around unnecessary objects.
+function onFinalize(obj, event) {
+  // You can do whatever you want with the object
+  obj.dispose();
+}
+
+function setup() {
+  // This object can be safely garbage collected,
+  // and the resulting shutdown function will not be called.
+  // There are no leaks.
+  const myDisposableObject = {
+    dispose() {
+      // Free your resources synchronously
+      stdout.write('disposed.\n');
+    },
+  };
+
+  finalization.register(myDisposableObject, onFinalize);
+}
+
+setup();
+```
+
+The code above relies on the following assumptions:
+
+* arrow functions are avoided
+* regular functions are recommended to be within the global context (root)
+
+Regular functions _could_ reference the context where the `obj` lives, making the `obj` not garbage collectible.
+
+Arrow functions will hold the previous context. Consider, for example:
```

Comment:
After some changes requested by Matteo, I thought it would be a good idea to document exactly how the user should use the feature and what they could do to accidentally hold the reference to the object.

What about:
```suggestion
Arrow functions will capture the `this` binding and possibly hold the reference to the `ref` object. Consider, for example:
```